gson/src/main/java/com/google/gson/stream/JsonWriter.java

```diff
@@ -217,39 +214,46 @@ public JsonWriter(Writer out) {
    * @param indent a string containing only whitespace.
    */
   public final void setIndent(String indent) {
-    if (indent.length() == 0) {
-      setFormattingStyle(null);
+    if (indent.isEmpty()) {
+      setFormattingStyle(FormattingStyle.COMPACT);
```

Comment:
Maybe the documentation should explain that this overrides any existing `FormattingStyle`, rather than just changing its indentation.